src/bob/resource/core.clj

```diff
@@ -0,0 +1,33 @@
+;   This file is part of Bob.
+;
+;   Bob is free software: you can redistribute it and/or modify
+;   it under the terms of the GNU General Public License as published by
+;   the Free Software Foundation, either version 3 of the License, or
+;   (at your option) any later version.
+;
+;   Bob is distributed in the hope that it will be useful,
+;   but WITHOUT ANY WARRANTY; without even the implied warranty of
+;   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+;   GNU General Public License for more details.
+;
+;   You should have received a copy of the GNU General Public License
+;   along with Bob. If not, see <http://www.gnu.org/licenses/>.
+
+(ns bob.resource.core
+  (:require [manifold.deferred :as d]
+            [bob.pipeline.internals :as p]
+            [bob.resource.internals :as r]))
+
+(defn mount-resource
+  "Creates the initial state of the build with a resource.
+
+  This works as follows:
+  - Downloads a zip file from the `url`.
+  - Expands it to a temp directory.
+  - Creates a container using the `image`.
+  - Copies over the contents to the dir `/tmp` inside the container.
+  - Creates an image from this container and returns its id."
+  [url image]
+  (d/let-flow [out-dir (r/fetch-resource url)
+               id      (r/initial-container-of out-dir image)]
+    (p/commit-to-image id [] [])))
```

Comment:
not sure if that will be better, but maybe it makes sense to make separate use case in the `commit-to-image` function, so it will work like `(p/commit-to-image id)`. in current case, I have to go back to the function in order to remember what were these variables.